detect/git.go

```diff
@@ -71,3 +86,94 @@ func (d *Detector) DetectGit(gitCmd *sources.GitCmd) ([]report.Finding, error) {
 	log.Debug().Msg("Note: this number might be smaller than expected due to commits with no additions")
 	return d.findings, nil
 }
+
+func (d *Detector) detectBinary(repo *git.Repository, commitHash string, filePath string) error {
+	logger := log.With().Str("commit", commitHash).Str("path", filePath).Logger()
+	// Check if the file is worth scanning.
+	if !shouldScanBinaryFile(filePath) {
+		logger.Debug().Msg("Skipping binary file.")
+		return nil
+	}
+	logger.Debug().Msg("Scanning binary file.")
+
+	// Read the file.
+	commit, err := repo.CommitObject(plumbing.NewHash(commitHash))
+	if err != nil {
+		return fmt.Errorf("failed to get commit object: %w", err)
+	}
+
+	// Get the tree associated with this commit
+	tree, err := commit.Tree()
+	if err != nil {
+		return fmt.Errorf("failed to get tree from commit: %w", err)
+	}
+
+	// Get the specific file in the commit's tree
+	file, err := tree.File(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to find file in tree: %w", err)
+	}
+
+	// Check if the file is a reasonable size.
+	if file.Size > d.MaxTargetMegaBytes {
+		logger.Debug().
+			Int64("size", file.Size).
+			Int64("limit", d.MaxTargetMegaBytes).
+			Msg("Skipping binary file: it exceeds size limit.")
+		return nil
+	}
+
+	// Open the file content as a reader
+	reader, err := file.Reader()
+	if err != nil {
+		return fmt.Errorf("failed to open file reader: %w", err)
+	}
+	defer func(reader io.ReadCloser) {
+		_ = reader.Close()
+	}(reader)
+
+	// Scan the reader.
+	findings, err := d.DetectReader(reader, 1*1024*1024)
```

Comment:
@zricethezav I'm trying to avoid duplicating code. Is there a meaningful difference between how [`DetectReader`](https://github.com/gitleaks/gitleaks/blob/c5b15c9d74036ca0c32134280c3d941b41e4f2d9/detect/reader.go#L11) and [`DetectFiles`](https://github.com/gitleaks/gitleaks/blob/c5b15c9d74036ca0c32134280c3d941b41e4f2d9/detect/directory.go#L42) work?

i.e., would it be reasonable for both `DetectGit` and `DetectFiles` to call `DetectReader` instead of implementing their own logic?